### PR TITLE
Fix: code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes

- Remove length check in favour of truthiness of the object
- Remove unnecessary use of comprehension
- Remove unnecessary comprehension
- Use identity check for comparison with True
- Pass string format arguments as logging method parameters

### CODE REVIEW

**Comment by nan-wang** [comment]
hi, @withshubh Thanks for your contributions. I'm afraid your commit messages do not comply with our guidelines.

Reference:
https://github.com/jina-ai/jina/pull/1949/checks?check_run_id=1909788648#step:7:13

Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)

**Comment by withshubh** [comment]
> hi, @withshubh Thanks for your contributions. I'm afraid your commit messages do not comply with our guidelines.
> 
> Reference:
> [#1949 (checks)](https://github.com/jina-ai/jina/pull/1949/checks?check_run_id=1909788648#step:7:13)
> 
> Bad commit message, please refer to:
> 
> * [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
> * [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md?rgh-link-date=2021-02-16T14%3A11%3A06Z)

Fixed it. :sparkles:

**Comment by florian-hoenicke** [comment]
@withshubh The branch has merge conflicts. Could you please resolve them?

**Comment by florian-hoenicke** [comment]
@withshubh there are still some issues with flake-8, docstrings and code format.
You need to install our pre-commit hook
```
pip install pre-commit
pre-commit install
```

File: jina/executors/decorators.py [hunk 1]
```diff
@@ -205,26 +205,28 @@ def _merge_results_after_batching(
     if not final_result:
         return
 
-    if isinstance(final_result[0], np.ndarray):
-        if len(final_result[0].shape) > 1:
+    if final_result and merge_over_axis is not None:
```
Comment (florian-hoenicke) [inline jina/executors/decorators.py:208]: the file has to be formatted with the pre-commit hook

**Review by florian-hoenicke — CHANGES_REQUESTED** [review]

File: jina/executors/decorators.py (hunk 1 above)
Comment (withshubh) [reply jina/executors/decorators.py:208]: on it :rocket:

**Comment by withshubh** [comment]
@florian-hoenicke done! :sparkles:

**Comment by nan-wang** [comment]
Ooops~ Bad commit message, please refer to:
- [Commit Message Guideline for the First Time Contributor](https://github.com/jina-ai/jina/issues/553)
- [Contributing Guideline](https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md)

**Pull request closed without merging**